### ISSUE TITLE
Use sub claim instead of identity for interoperatibility

### ISSUE THREAD

**Issue by psafont** [vimalloc/flask-jwt-extended#65]
Hello!

The library uses the claim `identity` for identifying the subject. This makes the library a bit awkard to interop with external identity providers or client implementations.

The JWT RFC recommends using `sub` for identifying the principal:

```
4.1.2. "sub" (Subject) Claim
The "sub" (subject) claim identifies the principal that is the
subject of the JWT. The claims in a JWT are normally statements
about the subject. The subject value MUST either be scoped to be
locally unique in the context of the issuer or be globally unique.
The processing of this claim is generally application specific. The
"sub" value is a case-sensitive string containing a StringOrURI
value. Use of this claim is OPTIONAL.
```

I was thinking about making the "identity" claim to be changeable with a config key. This would allow for several pro:
* Interoperability: allows the user to select a claim for "identity", allowing users of the library to interop with opinionated code.
* Smooth deprecation policy: default value can be left to be the same as now, while allowing users to change it; the default value can be changed whenever backwards-compatibility isn't a concern.

As for the cons, while decoding should be easy to change, encoding is another matter.

Thoughts?

**Comment by vimalloc** [comment]
Got this merged. Pypi changed how packages are uploaded out from under me, so I need to figure that out later tonight, then I'll get it setup there.

Cheers.

**Comment by vimalloc** [comment]
Released in 3.1.0
